packages/x-data-grid-generator/package.json

```diff
@@ -43,6 +43,8 @@
     "@mui/icons-material": "^5.16.14",
     "@mui/material": "^5.16.14",
     "@types/chance": "^1.1.6",
+    "react": "^19.0.0",
+    "react-dom": "^19.0.0",
```

Comment:
`@mui/x-data-grid-generator`, `@mui/x-internals`, and `@mui/x-license` packages were the culprits of resolving `react-dom` with react-dom@19 and `react-dom@18` when using for testing on react 18.
Explciitly adding a dev dependency avoids such issue and we no longer need to override the `auto-install-peers` setting to `false` to properly test on react 18. 👍

Reply:
This makes total sense, I did the same for the `@mui/material` dependency in https://github.com/mui/mui-x/pull/14142#discussion_r1711468731 to make version resolution work as expected 👍🏻